### DISCUSSION TITLE
How can use short type name?

### DISCUSSION THREAD

**Discussion by malegend** [sebastienros/jint#1332] [Q&A] [answered]
Can I write the call in a shorter way?
e.g.:
The original:
`var ListOfString=System.Collections.Generic.List(System.String);`
`var list=new ListOfString();`
OR
`var ListOf=System.Collections.Generic.List;`
`var string=System.String;`
`var list=new (ListOf(string))();`

Desired effect:
`var list=new ListOf(string)(32);`
or:
`var list=new ListOf(string,32);`

**Comment by lahma** [comment]
```c#
var engine = new Engine();
engine.SetValue("StringList", TypeReference.CreateTypeReference<List<string>>(engine));
Assert.Equal("a", engine.Evaluate("var list = new StringList(32); list.add('a'); return list[0];"));
```

**Comment by malegend** [comment, marked as answer]
But Not flexible enough
It is not possible to define a reference for each type in advance.
e.g.:

` List<string>,List<int>,List<double>...`

**Reply by lahma** [reply]
```c#
var engine = new Engine();
engine.SetValue("String", TypeReference.CreateTypeReference<string>(engine));
engine.SetValue("CreateList", (Type type) =>
{
    var listType = typeof(List<>).MakeGenericType(type);
    return Activator.CreateInstance(listType);
});
Assert.Equal("a", engine.Evaluate("var list = CreateList(String); list.add('a'); return list[0];"));
```

**Comment by malegend** [comment]
GREAT!
